pkg/controllers/nodeclaim/lifecycle/termination_test.go

```diff
@@ -353,25 +353,14 @@ var _ = Describe("Termination", func() {
 		}))
 	})
 	It("should not delete Nodes if the NodeClaim is not registered", func() {
-		ExpectApplied(ctx, env.Client, nodePool, nodeClaim)
-		ExpectObjectReconciled(ctx, env.Client, nodeClaimController, nodeClaim)
-
-		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
-		_, err := cloudProvider.Get(ctx, nodeClaim.Status.ProviderID)
-		Expect(err).ToNot(HaveOccurred())
-
 		node := test.NodeClaimLinkedNode(nodeClaim)
-		// Remove the unregistered taint to ensure the NodeClaim can't be marked as registered
-		node.Spec.Taints = nil
-		ExpectApplied(ctx, env.Client, node)
-		_ = ExpectObjectReconcileFailed(ctx, env.Client, nodeClaimController, nodeClaim)
-		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
-		Expect(nodeClaim.StatusConditions().Get(v1.ConditionTypeRegistered).IsFalse()).To(BeTrue())
+		ExpectApplied(ctx, env.Client, nodePool, nodeClaim, node)
```

Comment:
IMO: You should launch the NodeClaim at least but just not register it -- so have one reconciliation for the launch but don't do the other